src/domain/resource/resourceKeyboardReservation/ResourceKeyboardSlotPicker.js

```diff
@@ -0,0 +1,311 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import injectT from '../../../../app/i18n/injectT';
+import { getIsSlotReserved, getIsSlotInPast, getInMs } from './resourceKeyboardSlotPickerUtils';
+
+const rootClass = 'app-ResourcePageKeyboardTimePicker';
+
+function getClassName(className) {
+  return `${rootClass}__${className}`;
+}
+
+const classes = {
+  root: rootClass,
+  fieldGrid: getClassName('field-grid'),
+};
+
+function getTime(dateTime) {
+  const date = new Date(dateTime);
+  const hours = date.getHours();
+  const minutes = date.getMinutes();
+  const hoursWithPadding = String(hours).padStart(2, '0');
+  const minutesWithPadding = String(minutes).padStart(2, '0');
+
+  return `${hoursWithPadding}:${minutesWithPadding}`;
+}
+
+function getSlotCount(time, slotSize) {
+  const [timeHours, timeMinutes] = time.split(':');
+  const timeInMs = getInMs(timeHours, timeMinutes);
+  const [slotHours, slotMinutes] = slotSize.split(':');
+  const slotSizeInMs = getInMs(slotHours, slotMinutes);
+
+  return timeInMs / slotSizeInMs;
+}
+
+function labelStartTimeOptions(
+  slotsWithReservationStatus,
+  selectedBounds,
+  maxSlotCount,
+  minSlotCount,
+) {
+  // If there's no end bound any time is ok.
+  // If both bounds are set, any time are ok. When the use selects a
+  // bound, the companion bound is reset.
+  if (!selectedBounds.end || (selectedBounds.start && selectedBounds.end)) {
+    return slotsWithReservationStatus.map(slot => ({
+      ...slot,
+      isResultsInNegativeTime: false,
+      isResultsInOverlappingReservation: false,
+      isResultsInUnderMinTime: false,
+      isResultsInOverMaxTime: false,
+    }));
+  }
+
+  // eslint-disable-next-line max-len
+  const getIsResultsInNegativeTime = slotStart => new Date(slotStart).getTime() > new Date(selectedBounds.end).getTime();
```

Comment:
Will this cause time zone issues if the computer is set to Stockholm time? Could be better to use moment.tz?

Reply:
I'm not sure what issue you are referencing here. I tested it out with a Mumbai and Sao Paulo timezone to verify behaviour.

Generally speaking, I don't think that there should be huge danger for things going wrong because of timezones. JS dates try really hard to be timezone aware, in that sense it comes out of the box. From general code health, I think it's OK.

But there is the more usability oriented question which has to do with the behaviour. Currently times are shown in the locale time of the browser, which may actually be inoptimal. If you were referencing this, then you I agree with you. We could use `moment.tz` here to force the timezone to Helsinki time, which would allow visitors from abroad to make reservations in locale time, which maybe makes more sense. May be a bit overkill as well, I'll have to look into what it entails.